### DISCUSSION TITLE
is This posible ??? findAll(filter: object), dinamic filter ??

### DISCUSSION THREAD

**Discussion by emazzu** [prisma/prisma#17870] [Q&A]
Hi People. 
I need know, if is possible it ??

I want to call to findAll: 
http://localhost:3000/users/{where:{id:38}}

my controller: 
@Get('/:filter')
  findAll(@Param() params: {filter: object}): Promise<User[]> {
    console.log(params.filter)
    return this.usersService.findAll(params.filter);
  }

my service:
  findAll(filter: object) {
    /* return `This action returns all users`; */
    console.log(filter)
    return this.prisma.user.findMany(filter);
  }
  
but the response are all records, and it dont response by the where ?? 
it will possible call findAll with de json object ??? is very important.

I don't know what complete filter is going to come, where, order, limit, etc.

Regars
Eduardo

**Comment by nurul3101** [comment]
Hello @emazzu 👋 

Instead of getting the dynamic filter as a path parameter, can you try getting the filter in a query parameter?

So, for example, you can define the path as `http://localhost:3000/users?filter={where:{id:38}}` and then pass the filter query parameter to the prisma query?

**Reply by emazzu** [reply]
hi nuru3101, 
now I trying from postman, but then from angular, the user build de filter and I call endpoint with de filter. 

I like prisma, but I m in the beginning.

Thanks for your ansewing.

**Reply by emazzu** [reply]
but, yes, from angular, I will try to call an endpoint http://localhost:3000/users?filter={where:{id:38}, orderby.........}
Something like this,

**Reply by nurul3101** [reply]
Sure, did passing the filter worked as expected?

**Reply by emazzu** [reply]
hi nuru, how are you, well, i did the test, I just had time today.

I dont know, what i m doing wrong !!!

controller
```
@Get()
  async findAll(@Query() query: {where?: object, orderBy?: object}): Promise<User[]> {
    return this.usersService.findAll(query.where);
  }
```
service
```
findAll(where?: object) {
    console.log(where)
    return this.prisma.user.findMany({where: where});
}
```
call
`http://localhost:3000/users?where={where:{id:38}}`

response
```
{
  "statusCode": 500,
  "message": "Internal server error"
}
```
error
```
Error: 
Invalid `this.prisma.user.findMany()` invocation in
C:\Users\eduardo_mazzucchelli\BKE\nest-prisma\src\users\users.service.ts:16:29

  13
  14 findAll(where?: object) {
  15   console.log(where)
→ 16   return this.prisma.user.findMany({
         where: '{where:{id:38}}'
                ~~~~~~~~~~~~~~~~~
       })

Argument where: Got invalid value '{where:{id:38}}' on prisma.findManyuser. Provided String, expected userWhereInput:

```

**Reply by nurul3101** [reply]
You are passing a string in the findMany `where` clause. 
Your actual query should look like this:
```typescript
this.prisma.user.findMany({
         where: {
           id:38
         }  
    })
```
You would need to parse the where string that you are getting and construct a query similar to this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
